lib/common/http2client.c

```diff
@@ -1156,6 +1173,9 @@ static void on_write_complete(h2o_socket_t *sock, const char *err)
             H2O_STRUCT_FROM_MEMBER(struct st_h2o_http2client_stream_t, output.sending_link, link);
         h2o_linklist_unlink(link);
 
+        if (stream->_use_expect && stream->state.req == STREAM_STATE_BODY)
+            continue;
```

Comment:
I wonder if we need to arm first-byte timeout like we do below, considering that we will be waiting for the server to send a 100.

Otherwise, the request might get stuck forever if the connection is kept busy by other requests.

Reply:
Nice catch! Done in 5f1d2d4